tidy3d/components/scene.py

```diff
@@ -1112,18 +1118,30 @@ def _pcolormesh_shape_custom_medium_structure_eps(
                             axis=normal_axis_ind, pos=normal_position
                         )
                 else:
-                    eps_mean = (eps_diag[0] + eps_diag[1] + eps_diag[2]) / 3
+                    if eps_component == "xx":
```

Comment:
There's some logic duplication below. I think similar to the comp2ind you had earlier, you can define one for the double index components 

```
comp2ind = {dim+dim: index for dim, index in zip('xyz', range(3))}
```

and then,

```py
if eps_component is None:
    eps = ...
elif eps_component in comp2ind:
    eps = eps_diag[comp2ind[eps_component]]
else
    ....
```